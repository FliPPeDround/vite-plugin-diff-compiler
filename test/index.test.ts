import { describe, expect, test } from 'vitest'
import {
  diffCompiler,
} from '../package/core/src/diffCompiler'

describe('fixtures', async () => {
  const files = import.meta.glob('./fixtures/*.{vue,js,ts}', {
    eager: true,
    as: 'raw',
  })

  for (const [id, code] of Object.entries(files)) {
    test(id.replace(/\\/g, '/'), async () => {
      // 获取文件名
      const ext = id.split('.').at(-2)!.split('/').pop()!

      const exec = () => diffCompiler(code, ext)
      if (id.includes('error'))
        expect(exec).toThrowErrorMatchingSnapshot()

      else
        expect(exec()).toMatchSnapshot()
    })
  }
})
