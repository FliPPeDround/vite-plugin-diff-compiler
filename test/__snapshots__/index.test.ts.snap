// Vitest Snapshot v1

exports[`fixtures > ./fixtures/basic.js 1`] = `
"

// #diff-compiler-start:basic
console.log('Hello basic')
// #diff-compiler-end:basic
"
`;

exports[`fixtures > ./fixtures/vue3.vue 1`] = `
"<script lang=\\"ts\\" setup>


// #diff-compiler-start:vue3
console.log('Hello vue3')
// #diff-compiler-end:vue3
</script>

<template>

  <!-- #diff-compiler-start:vue3 -->
    <h1>vue3</h1>
  <!-- #diff-compiler-end:vue3 -->
</template>

<style>

/* #diff-compiler-start:vue3 */
h1 {
  color: blue;
}
/* #diff-compiler-end:vue3 */
</style>
"
`;
